kcp/utils.sh

```diff
@@ -167,3 +167,80 @@ registerSyncTarget() {
     echo "Waiting for the SyncTarget resource to reach ready state"
     KUBECONFIG="${CPS_KUBECONFIG}" kubectl wait --for=condition=Ready=true synctarget/"${WORKLOAD_CLUSTER}" --timeout 3m
 }
+
+createAPIBinding() {
+    exportName=$1
+    KUBECONFIG="${CPS_KUBECONFIG}" kubectl kcp ws
+    url=$(KUBECONFIG="${CPS_KUBECONFIG}" kubectl get workspace $SERVICE_WS -o jsonpath='{.status.URL}')
+    path=$(basename $url)
+    KUBECONFIG="${CPS_KUBECONFIG}" kubectl kcp ws use $USER_WS
+
+    permissionClaims='
+  permissionClaims:
+  - group: ""
+    resource: "secrets"
+    state: "Accepted"
+  - group: ""
+    resource: "namespaces"
+    state: "Accepted"'
+
+  if [ exportName == "gitopsrvc-appstudio-shared" ]; then
+    permissionClaims="${acceptedPermissionClaims}
+  - group: \"managed-gitops.redhat.com\"
+    resource: \"gitopsdeployments\"
+    state: \"Accepted\"
+    identityHash: ${identityHash}"
+  fi
+
+cat <<EOF | KUBECONFIG="${CPS_KUBECONFIG}" kubectl apply -f -
+apiVersion: apis.kcp.dev/v1alpha1
+kind: APIBinding
+metadata:
+  name: ${exportName}
+spec:
+${permissionClaims}
+  reference:
+    workspace:
+      path: ${path}
+      exportName: ${exportName}
+EOF
+}
+
+permissionToBindAPIExport() {
+  # Create permissions to bind APIExports. We need this workaround until KCP fixes the bug in their admission logic. Ref: https://github.com/kcp-dev/kcp/issues/1939  
+  KUBECONFIG="${CPS_KUBECONFIG}" kubectl kcp ws
+  bindingName=$(KUBECONFIG="${CPS_KUBECONFIG}" kubectl get clusterrolebinding | grep $SERVICE_WS | awk '{print $1}')
+  userName=$(KUBECONFIG="${CPS_KUBECONFIG}" kubectl get clusterrolebindings $bindingName -o jsonpath='{.subjects[0].name}')
+
+  KUBECONFIG="${CPS_KUBECONFIG}" kubectl kcp ws use $SERVICE_WS
+
+cat <<EOF | KUBECONFIG="${CPS_KUBECONFIG}" kubectl apply -f -
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+  name: bind-apiexport
+rules:
+- apiGroups:
+  - apis.kcp.dev
+  resourceNames:
+  - gitopsrvc-backend-shared
```

Comment:
I see you are using these (`gitopsrvc-backend-shared` and `gitopsrvc-appstudio-shared`) a lot. Create a variable or constant instead of hardcoding them in several places.

Reply:
It was used in two places, so didn't give it much thought, I'll address this in the next story/PR which might introduce more cleanup with added functionality. In case there are too many>2 :P I'll update it up, it wouldn't hurt to use a var instead of it, the var name might be more confusing and long so wanted to keep it alike for ease fow now! Hope it's okay.